tiny-composer-installer.php

```diff
@@ -130,12 +131,12 @@ function select_version($versions, $channel)
     return $version[0];
 }
 
-function write_phar($phar)
+function write_phar($phar, $default_filename)
 {
-    $file = ($_SERVER['argc'] >= 2) ? $_SERVER['argv'][1] : tempnam(sys_get_temp_dir(), 'composer-');
+    $file = ($_SERVER['argc'] >= 2) ? $_SERVER['argv'][1] : $default_filename;
     if (!file_put_contents($file, $phar)) {
         throw new \RuntimeException("could not write PHAR to $file");
     }
     chmod($file, 0755);
```

Comment:
Check the return value of `chmod` and at least print warning.

Reply:
Fixed.